contracts/DxMgnPool.sol

```diff
@@ -57,8 +57,27 @@ contract DxMgnPool {
         require(depositToken.transferFrom(msg.sender, address(this), amount), "Failed to transfer deposit");
     }
 
+    function withdraw() public {
+        require(hasPoolingEnded(), "Pooling period is not over, yet");
+        
+        uint totalDepositAmount = 0;
+        uint totalMgnClaimed = 0;
+        Participation[] memory participations = participationsByAddress[msg.sender];
+        for (uint i = 0; i < participations.length; i++) {
+            totalDepositAmount += participations[i].amount;
```

Comment:
We need to weight the amount by the supply in the contract. Because we might lose some funds on the way.